src/components/common/Map.js

```diff
@@ -87,11 +94,29 @@ class Map extends React.Component {
   }
 
   onMapClick(e) {
-    let lat = parseFloat(e.latLng.lat().toFixed(6));
-    let lng = parseFloat(e.latLng.lng().toFixed(6));
-    this.props.updateTargetInfo("lat", lat);
-    this.props.updateTargetInfo("lng", lng);
-    this.props.updateTargetInfo("isVisible", true);
+    const lat = parseFloat(e.latLng.lat().toFixed(6));
+    const lng = parseFloat(e.latLng.lng().toFixed(6));
+    const { updateTargetInfo, unselectTarget, changeContent } = this.props;
+    unselectTarget();
+    updateTargetInfo("lat", lat);
```

Comment:
Here you have 3 calls changing the state of the app, which probably results in 3 re-renders. What about refactoring this to a `updateTargetInfo(target)` function where target is an object containing the `lat`, `lon` and `isVisible` properties?